conans/client/package_layouts/__init__.py

```diff
@@ -0,0 +1,44 @@
+# coding=utf-8
+
+import os
+
+from conans.errors import ConanException
+from conans.model.ref import ConanFileReference
+from conans.client.package_layouts.package_cache_layout import PackageCacheLayout
+from conans.client.package_layouts.package_user_layout import PackageUserLayout, LINKED_FOLDER_SENTINEL
+from conans.paths import is_case_insensitive_os
+
+if is_case_insensitive_os():
+    def check_ref_case(conan_reference, store_folder):
+        if not os.path.exists(store_folder):
+            return
+
+        tmp = store_folder
+        for part in conan_reference:
+            items = os.listdir(tmp)
+            try:
+                idx = [item.lower() for item in items].index(part.lower())
+                if part != items[idx]:
+                    raise ConanException("Requested '%s' but found case incompatible '%s'\n"
+                                         "Case insensitive filesystem can't manage this"
+                                         % (str(conan_reference), items[idx]))
+                tmp = os.path.normpath(tmp + os.sep + part)
+            except ValueError:
+                return
+else:
+    def check_ref_case(conan_reference, store_folder):  # @UnusedVariable
+        pass
+
+
+def get_package_layout(store_folder, conan_reference, short_paths=False):
```

Comment:
This, so far, belongs to the client_cache (base_folder only applies to PackageCacheLayout)

Reply:
What do you mean by `base_folder`? 

If you mean `store_folder`, I need it to reach the _sentinel_ file that stores where the package is in the developer workspace.

Reply:
Ok, moved to `SimplePaths` (it is the ancestor of `ClientCache`)